frontend/src/components/page-events/event-processing-util.js

```diff
@@ -0,0 +1,126 @@
+
+const STRAPI_RECURRING_TIME = "event-times.recurring-time";
+const STRAPI_SINGLE_TIME = "event-times.single-time";
+
+
+const addDays = (date, days) => {
+  const result = new Date(date);
+  result.setDate(result.getDate() + days);
+  return result;
+};
+
+const addWeeks = (date, weeks) => {
+  const result = new Date(date);
+  result.setDate(result.getDate() + weeks * 7);
+  return result;
+};
+
+const addMonths = (date, months) => {
+  const result = new Date(date);
+  result.setMonth(result.getMonth() + months);
+  return result;
+};
+
+const addYears = (date, years) => {
+  const result = new Date(date);
+  result.setFullYear(result.getFullYear() + years);
+  return result;
+};
+
+/*
+  * Returns whether the date is before today.
+  * We don't want events to disappear immediately after they start.
+  * We'll use this function to keep showing events until the end of the same day.
+ */
+const isTodayOrAfter = (date) => {
+  const copyOfDate = new Date(date);
+  const today = new Date();
+  today.setHours(0, 0, 0, 0);
+
+  return copyOfDate > today;
+};
+
+const DATE_ITERATOR = {
+  "Day": addDays,
+  "Week": addWeeks,
+  "Month": addMonths,
+  "Year": addYears,
+};
+
+const generateRecurringEvent = (timeInfo, baseEvent, maxOccurrences) => {
```

Comment:
Will this code work once time has passed but no new build has happened? I think maxOccurences and ShowIfPast need to be client side logic which filters the entire event list that is built at build time. 

For example, if we build website today with these parameters, but by next week nobody has updated the website, it won't get rebuilt, but we still want the one that past to disappear and another one from the future recurring to appear. 

Probably having server generate a default of max 20(?) future events is sufficient, and we can just have the clientside wittle that down as time passes. Am I making sense?

Reply:
So for what it's worth: During development, I at some point had events disappear as soon as the start date passed. I wanted to test what you're talking about here, so I created an event with a start time a few minutes in the future and built the site. Then I waited until that time was past (without rebuilding the site), and the event still disappeared once the time was past.

Another test is adding a `const date = new Date();` and `<div>{date.toString()}</div>` in the render method of a page to print a datestring to the page. The date updates on page refresh.

My interpretation of this was that the javascript was still being run on render (i.e., that it _is_ client side already). Unless this only happens in development and not in prod?

Reply:
I see. Let's get this checked in and see how prod responds to it then. If we need to iterate, that's fine